azavea/iow-boundary-tool
# Implement boundary submission form

## Overview

Create the submission and confirmation modals that allow the contributor to submit their work. Wire them up with the back end.

Closes #132

### Demo

https://user-images.githubusercontent.com/38668450/197290056-de35c42c-7b2e-4332-8d20-25f4c2ab9dff.mp4


### Notes
- I haven't implemented the download submission button yet, but I might be able to do so after #143.
- I cherry-picked b592194f06539ef94d5e102666d9ccc7dad5199e from #143, as I needed the utility city.
- Happy to decompose into smaller units, just wanted to open for review

## Testing Instructions

- scripts/resetdb
- scripts/server
- Login as c1@azavea.com
- You should be routed automatically to a draft
  - [x] back button works
- Review and submit, and add a note
  - [x] Ensure note is saved and metadata updated with user, timestamp
  - [x] Ensure sign out button signs out
  - [x] Ensure status button navigates to /submissions
- Test again (use resetdb) with an empty note
- Test again (use resetdb) with invalid permissions e.g. v1@azavea.com, an error toast should appear

## Checklist
